server/auth/types/openid/openid_auth.ts

```diff
@@ -272,7 +272,7 @@ export class OpenIdAuthentication extends AuthenticationType {
       return false;
     }
 
-    if (cookie.expiryTime > Date.now()) {
+    if (cookie.credentials.expiryTime > Date.now()) {
```

Comment:
Can you add a little more explanation on how setting expiryTime directly from credentials would resolve the issue at hand?

Reply:
OIDC response has two JWT token, ID Token and Refresh token. You are taking expiry time of ID token and setting it   as expiry of cookie, ID token are short lived token and once they are expired we can get new token using Refresh token, code for this is present in isValidCookie. 
In login flow, I have set ID token expiry value in 'cookie.credentials.expiry' and checking if ID token is still valid or not. If it is valid then true is returned, if it is not valid a new ID token will be fetched using Refresh token and value of this token are updated in cookie again.
User is logged-out only if both ID token and refresh token are expired or cookie is invalidated.